doc/markup_reference.rb

```diff
@@ -0,0 +1,917 @@
+#   #
+#   # -- Also not a horizontal rule.
+#   #
+#   # ---
+#
+# Rendered HTML:
+#
+# ------
+# Stuff between.
+#
+# \--- Not a horizontal rule.
+#
+# -- Also not a horizontal rule.
+#
+# ---
+#
+# === Text Markup
+#
+# Text in a paragraph, list item (any type), or heading
+# may have markup formatting.
+#
+# ==== Italic
+#
+# A single word may be italicized by prefixed and suffixed underscores.
+#
+# Examples:
+#
+#   # _Word_ in paragraph.
+#   # - _Word_ in bullet list item.
+#   # 1. _Word_ in numbered list item.
+#   # a. _Word_ in lettered list item.
+#   # [_word_] _Word_ in labeled list item.
+#   # ====== _Word_ in heading
+#
+# Any text may be italicized via HTML tag +i+ or +em+.
+#
+# Examples:
+#
+#   # <i>Two words</i> in paragraph.
+#   # - <i>Two words</i> in bullet list item.
+#   # 1. <i>Two words</i> in numbered list item.
+#   # a. <i>Two words</i> in lettered list item.
+#   # [<i>Two words</i>] <i>Two words</i> in labeled list item.
+#   # ====== <i>Two words</i> in heading
+#
+# ==== Bold
+#
+# A single word may be made bold by prefixed and suffixed asterisks.
+#
+# Examples:
+#
+#   # *Word* in paragraph.
+#   # - *Word* in bullet list item.
+#   # 1. *Word* in numbered list item.
+#   # a. *Word* in lettered list item.
+#   # [*word*] *Word* in labeled list item.
+#   # ====== *Word* in heading
+#
+# Any text may be made bold via HTML tag +b+.
+#
+# Examples:
+#
+#   # <b>Two words</b> in paragraph.
+#   # - <b>Two words</b> in bullet list item.
+#   # 1. <b>Two words</b> in numbered list item.
+#   # a. <b>Two words</b> in lettered list item.
+#   # [<b>Two words</b>] <b>Two words</b> in labeled list item.
+#   # ====== <b>Two words</b> in heading
+#
+# ==== Monofont
+#
+# A single word may be made monofont -- sometimes called "typewriter font" --
+# by prefixed and suffixed plus-signs.
+#
+# Examples:
+#
+#   # +Word+ in paragraph.
+#   # - +Word+ in bullet list item.
+#   # 1. +Word+ in numbered list item.
+#   # a. +Word+ in lettered list item.
+#   # [+word+] +Word+ in labeled list item.
+#   # ====== +Word+ in heading
+#
+# Any text may be made monofont via HTML tag +tt+ or +code+.
+#
+# Examples:
+#
+#   # <tt>Two words</tt> in paragraph.
+#   # - <tt>Two words</tt> in bullet list item.
+#   # 1. <tt>Two words</tt> in numbered list item.
+#   # a. <tt>Two words</tt> in lettered list item.
+#   # [<tt>Two words</tt>] <tt>Two words</tt> in labeled list item.
+#   # ====== <tt>Two words</tt> in heading
+#
+# ==== Character Conversions
+#
+# Certain combinations of characters may be converted to special characters;
+# whether the conversion occurs depends on whether the special character
+# is available in the current encoding.
+#
+# - <tt>(c)</tt> converts to (c) (copyright character); must be lowercase.
+#
+# - <tt>(r)</tt> converts to (r) (registered trademark character); must be lowercase.
+#
+#  - <tt>'foo'</tt> converts to 'foo' (smart single-quotes).
+#    [Reviewers: This did not work; should it?]
```

Comment:
(This is me sign-posting where there are direct questions in this diff, to try making it easier to locate them. I'll quote the question text.)

> [Reviewers: This did not work; should it?]